### ISSUE TITLE
Package updates for v4

### ISSUE THREAD

**Issue by veggiesaurus** [CARTAvis/carta-frontend#1974]
Various package updates for v4 are required. We should aim to be compatible with the latest stable versions of our dependencies for beta-1. Excluding Blueprint4 (which has a separate issue defined #1395), `npm outdated` shows the following (the commented-out ones should be trivial to update):

```shell
Package                        Current    Wanted    Latest  Location                                   Depended by
#@testing-library/jest-dom       5.14.1    5.16.5    5.16.5  node_modules/@testing-library/jest-dom     carta-frontend
#@testing-library/react          12.0.0    12.1.5    13.3.0  node_modules/@testing-library/react        carta-frontend
#@testing-library/user-event     13.1.9    13.5.0    14.4.3  node_modules/@testing-library/user-event   carta-frontend
#@types/gapi                     0.0.40    0.0.40    0.0.42  node_modules/@types/gapi                   carta-frontend
#@types/gapi.auth2               0.0.55    0.0.55    0.0.56  node_modules/@types/gapi.auth2             carta-frontend
#@types/jest                    26.0.24   26.0.24    28.1.6  node_modules/@types/jest                   carta-frontend
#@types/jquery                    3.5.6    3.5.14    3.5.14  node_modules/@types/jquery                 carta-frontend
#@types/lodash                 4.14.171  4.14.182  4.14.182  node_modules/@types/lodash                 carta-frontend
#@types/node                     16.3.2  16.11.48    18.7.3  node_modules/@types/node                   carta-frontend
@types/plotly.js               1.54.11   1.54.22    2.12.2  node_modules/@types/plotly.js              carta-frontend
#@types/prismjs                  1.16.6    1.26.0    1.26.0  node_modules/@types/prismjs                carta-frontend
@types/react                   17.0.14   17.0.48   18.0.17  node_modules/@types/react                  carta-frontend
#@types/react-color               3.0.5     3.0.6     3.0.6  node_modules/@types/react-color            carta-frontend
#@types/react-dom                17.0.9   17.0.17    18.0.6  node_modules/@types/react-dom              carta-frontend
#@types/react-plotly.js           2.5.0     2.5.1     2.5.1  node_modules/@types/react-plotly.js        carta-frontend
#@types/react-window              1.8.4     1.8.5     1.8.5  node_modules/@types/react-window           carta-frontend
ajv                              8.6.1    8.11.0    8.11.0  node_modules/ajv                           carta-frontend
#axios                           0.21.2    0.21.4    0.27.2  node_modules/axios                         carta-frontend
chart.js                         3.7.1     3.9.1     3.9.1  node_modules/chart.js                      carta-frontend
golden-layout                    1.5.9     1.5.9     2.5.0  node_modules/golden-layout                 carta-frontend
idb                              6.1.3     6.1.5     7.0.2  node_modules/idb                           carta-frontend
konva                            8.1.1    8.3.11    8.3.11  node_modules/konva                         carta-frontend
#mnemonist                       0.38.3    0.38.5    0.39.2  node_modules/mnemonist                     carta-frontend
mobx                             6.3.2     6.6.1     6.6.1  node_modules/mobx                          carta-frontend
mobx-react                       7.2.0     7.5.2     7.5.2  node_modules/mobx-react                    carta-frontend
plotly.js                       1.58.5    1.58.5    2.14.0  node_modules/plotly.js                     carta-frontend
#prettier                         2.6.2     2.6.2     2.7.1  node_modules/prettier                      carta-frontend
#prismjs                         1.27.0    1.28.0    1.28.0  node_modules/prismjs                       carta-frontend
protobufjs                       6.9.0     6.9.0     7.0.0  node_modules/protobufjs                    carta-frontend
react                           17.0.2    17.0.2    18.2.0  node_modules/react                         carta-frontend
react-chartjs-2                  3.1.1     3.3.0     4.3.1  node_modules/react-chartjs-2               carta-frontend
react-dom                       17.0.2    17.0.2    18.2.0  node_modules/react-dom                     carta-frontend
react-konva                   17.0.2-4  17.0.2-6    18.2.1  node_modules/react-konva                   carta-frontend
react-resize-detector            6.7.4     6.7.8     7.1.2  node_modules/react-resize-detector         carta-frontend
react-rnd                        9.2.0     9.2.0    10.3.7  node_modules/react-rnd                     carta-frontend
react-scripts                    4.0.3     4.0.3     5.0.1  node_modules/react-scripts                 carta-frontend
#react-scroll-to-bottom           4.1.2     4.2.0     4.2.0  node_modules/react-scroll-to-bottom        carta-frontend
react-simple-code-editor        0.11.0    0.11.3    0.11.3  node_modules/react-simple-code-editor      carta-frontend
react-virtualized-auto-sizer     1.0.5     1.0.6     1.0.6  node_modules/react-virtualized-auto-sizer  carta-frontend
#react-window                     1.8.6     1.8.7     1.8.7  node_modules/react-window                  carta-frontend
#rxjs                             7.2.0     7.5.6     7.5.6  node_modules/rxjs                          carta-frontend
#sass                            1.52.3    1.54.4    1.54.4  node_modules/sass                          carta-frontend
#typescript                       4.3.5     4.7.4     4.7.4  node_modules/typescript                    carta-frontend
```

**Comment by veggiesaurus** [comment]
In addition to the above updates, we need to update our WebAssembly dependencies (including the emscripten docker version)

**Comment by YuHsuan-Hwang** [comment]
For `react-scripts`, upgrading from v4 to v5 breaks the builds. Will need further investigation.
```
Creating an optimized production build...
Failed to compile.

Module not found: Error: Can't resolve 'path' in 'PATH/carta-frontend/wasm_src/ast_wrapper/build'
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
...
```

**Comment by kswang1029** [comment]
@YuHsuan-Hwang @veggiesaurus advices for this issue? Done? Left over? Close and create narrow-down new ones?

**Comment by YuHsuan-Hwang** [comment]
@kswang1029 I think we can leave it open. React, testing-library, and Konva upgrades are blocked by Blueprintjs upgrade. There are also other new outdated packages.
